### PR TITLE
address warnings in Routing manager

### PR DESCRIPTION
**Pull request by kianzarrin**
Note: the code is functionally the same. just resolved some warnings and renamed some variables. its easier to follow the commits for individual changes.

- in routing manager I can understand why we refer to next/prev segment as such but I don't understand why we call the "common node" the "next node". I decided to just call it node to avoid confusion.
- removed duplicated variables and added doc

also fixed these warnings:
- commented out code should not exist (if necessary then create issue | if not necessary then delete).
- avoid using obsolete code.
- observe order of members a little bit but also turned off the warning because we usually don't observe it.
- use `new ()` to instantiate 
- use inline struct initializer.
- turned off some warnings that I deemed unnecessary because we usually don't follow it anyway.

I think we should not have warning rules that are usually broken because too many warnings means that we will ignore them. If we do have a warning it should be consistent with code cleanup tool.

~~I made this a draft because I had some questions.~~

### CODE REVIEW

File: TLM/TLM/Manager/Impl/RoutingManager.cs [hunk 1]
```diff
@@ -19,10 +19,10 @@ public class RoutingManager
         : AbstractGeometryObservingManager,
           IRoutingManager
     {
-        private RoutingManager() { }
-
         public static readonly RoutingManager Instance = new RoutingManager();
 
+        private RoutingManager() { }
```
Comment (kianzarrin) [inline TLM/TLM/Manager/Impl/RoutingManager.cs:24]: Do we need this warnings? should I move the constructor down? I think its better to get rid of the warning. It makes sense to keep the constructor close to Instance.
image

File: TLM/TLM/Manager/Impl/RoutingManager.cs [hunk 2]
```diff
@@ -19,10 +19,10 @@ public class RoutingManager
...
+
         private const NetInfo.LaneType ROUTED_LANE_TYPES =
```
Comment (kianzarrin) [inline TLM/TLM/Manager/Impl/RoutingManager.cs:26]: what about this one?
image

Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/RoutingManager.cs:26]: why is this warning important?
image

Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/RoutingManager.cs:26]: this warning seems unnecessary:
image

**Comment by kianzarrin** [comment]
github is not working?

**Review by aubergine10 — APPROVED** [review]
LGTM 👍

File: TLM/TLM/Manager/Impl/RoutingManager.cs [hunk 3]
```diff
@@ -396,62 +395,63 @@ public class RoutingManager
             ExtSegment prevSeg = Constants.ManagerFactory.ExtSegmentManager.ExtSegments[segmentId];
             ExtSegmentEnd prevEnd = segEndMan.ExtSegmentEnds[segEndMan.GetIndex(segmentId, startNode)];
 
-            ushort prevSegmentId = segmentId;
-            int prevLaneIndex = laneIndex;
+            ushort prevSegmentId = segmentId; // this variable is duplicate (prevSegmentId == segmentId always)
+            int prevLaneIndex = laneIndex; // this variable is duplicate (prevLaneIndex == laneIndex always)
             uint prevLaneId = laneId; // this variable is duplicate (prevLaneId == laneId always)
```
Comment (kianzarrin) [inline TLM/TLM/Manager/Impl/RoutingManager.cs:400]: is there any better way of dealing with these duplicate variables?
is there any way of marking that as read-only or something?
should I change parameter names or would that make the method confusing?

Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/RoutingManager.cs]: I just renamed parameters. we have enough variables in this method. we don't need duplicates!

**Comment by kianzarrin** [comment]
sorry @aubergine10 I did a few more renames/docs

**Comment by aubergine10** [comment]
> sorry @aubergine10 I did a few more renames/docs

Yeah, I've been following the commits - still looks good. Definite improvement in legibility. 👍

**Review by krzychu124 — APPROVED** [review]

**Pull request merged**